### PR TITLE
Add section about ALPN and HTTP tunnelling to binary protocol reference docs

### PR DESCRIPTION
**Pull request by quinchs**

### CODE REVIEW

**Review by 1st1 — APPROVED** [review]

File: docs/reference/protocol/index.rst [hunk 1]
```diff
@@ -18,6 +18,72 @@ clients and servers.  The protocol is supported over TCP/IP.
...
+``SCRAM-SHA-256``. If the auth method differs from the requested method, 
+the client should abort the authentication attempt.
+
+Once the :ref:`ref_authentication` phase is complete, the final responses' 
+body will contain an authorization token used to authenticate the HTTP 
+connection. The client then sends any following message to ``/db/{DATABASE}`` 
+with the following headers:
+
+* ``X-EdgeDB-User``: The username specified in the 
+  :ref:`ref_reference_connection`.
```
Comment (raddevon) [inline docs/reference/protocol/index.rst:74]:
```suggestion
  :ref:`connection parameters <ref_reference_connection>`.
```

File: docs/reference/protocol/index.rst [hunk 2]
```diff
@@ -18,6 +18,72 @@ clients and servers.  The protocol is supported over TCP/IP.
...
+
+* ``Authorization``: The authorization token received from the 
+  :ref:`ref_authentication` phase, prefixed by ``Bearer``.
```
Comment (raddevon) [inline docs/reference/protocol/index.rst:77]:
```suggestion
  :ref:`authentication <ref_authentication>` phase, prefixed by ``Bearer``.
```

File: docs/reference/protocol/index.rst [hunk 3]
```diff
@@ -18,6 +18,72 @@ clients and servers.  The protocol is supported over TCP/IP.
...
+
+* ``Content-Type``: Always ``application/x.edgedb.v_1_0.binary``.
+
+The response should be checked to match the content type, and the body should 
+be  parsed as the :ref:`ref_message_format` described below; multiple message 
+can be included in the response body, and should be parsed in order.
```
Comment (raddevon) [inline docs/reference/protocol/index.rst:83]:
```suggestion
be  parsed as the :ref:`message format <ref_message_format>` described below;
multiple message can be included in the response body, and should be parsed in
order.
```

**Review by raddevon — APPROVED** [review]
This is great, @quinchs! I found some very minor things. Otherwise, it looks good.

I did go back and provide alternative labels for most of the references where I didn't think they should be capitalize. Please correct me if I'm mistaken on those.

File: docs/reference/protocol/index.rst [hunk 4]
```diff
@@ -18,6 +18,74 @@ clients and servers.  The protocol is supported over TCP/IP.
     dataformats
 
 
+.. _ref_protocol_connecting:
+
+Connecting to EdgeDB 
+====================
+
+The EdgeDB binary protocol has two modes of operation over TCP/IP:
```
Comment (elprans) [inline docs/reference/protocol/index.rst:26]:
```suggestion
The EdgeDB binary protocol has two modes of operation over TCP/IP: 
```

Technically, HTTP isn't always TCP (HTTP/3 is UDP, for example), so I'd reword this.

Comment (elprans) [reply docs/reference/protocol/index.rst:26]: No, the regular socket binary protocol is TCP.

**Review by elprans — APPROVED** [review]

File: docs/reference/protocol/index.rst [hunk 5]
```diff
@@ -18,6 +18,74 @@ clients and servers.  The protocol is supported over TCP/IP.
...
+  :ref:`authentication <ref_authentication>` phase, prefixed by ``Bearer``.
+
+* ``Content-Type``: Always ``application/x.edgedb.v_1_0.binary``.
+
+The response should be checked to match the content type, and the body should 
+be parsed as the :ref:`message format <ref_message_format>` described below;
+multiple message can be included in the response body, and should be parsed in
+order.
+
+.. _ALPN Protocol: https://github.com/edgedb/rfcs/blob/master/text/1008-tls-and-alpn.rst#alpn-and-protocol-changes
```
Comment (elprans) [inline docs/reference/protocol/index.rst:87]:
```suggestion
.. _ALPN Protocol: 
    https://github.com/edgedb/rfcs/blob/master/text/
    1008-tls-and-alpn.rst#alpn-and-protocol-changes
```
